### PR TITLE
Don't use JObject.Merge in OpenId module

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

**Comment by hishamco** [comment]
Oops seems one of the unit tests has broken :(

File: src/OrchardCore.Modules/OrchardCore.OpenId/Deployment/OpenIdServerDeploymentSource.cs [hunk 1]
```diff
@@ -55,15 +55,10 @@ protected override async Task ProcessAsync(OpenIdServerDeploymentStep step, Depl
             RequireProofKeyForCodeExchange = settings.RequireProofKeyForCodeExchange,
         };
 
-        // Use nameof(OpenIdServerSettings) as name,
-        // to match the recipe step.
-        var obj = new JsonObject
+        result.Steps.Add(new JsonObject
         {
             ["name"] = nameof(OpenIdServerSettings),
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Deployment/OpenIdServerDeploymentSource.cs:60]: name should be `Settings`

File: src/OrchardCore.Modules/OrchardCore.OpenId/Deployment/OpenIdValidationDeploymentSource.cs [hunk 2]
```diff
@@ -19,12 +19,10 @@ protected override async Task ProcessAsync(OpenIdValidationDeploymentStep step,
...
-        // The 'name' property should match the related recipe step name.
-        var jObject = new JsonObject { ["name"] = nameof(OpenIdValidationSettings) };
-
-        // Merge settings as the recipe step doesn't use a child property.
-        jObject.Merge(JObject.FromObject(validationSettings));
-
-        result.Steps.Add(jObject);
+        result.Steps.Add(new JsonObject
+        {
+            ["name"] = nameof(OpenIdValidationSettings),
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Deployment/OpenIdValidationDeploymentSource.cs:24]:
```suggestion
            ["name"] = "Settings",
```

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]
Please be sure to test out changes before submitting a PR to ensure the changes actually solve the problem. As of this review, I am confident that the issue still exists.

**Comment by MikeAlhayek** [comment]
https://github.com/OrchardCMS/OrchardCore/pull/16862#issuecomment-2406177957

It is a test bug because now you changed how we export and import the settings "it was not working before" so the test was wrong

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**